### PR TITLE
MM-12420: Avoid slash duplications on gitlab urls configuration

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Avoid slash duplications on gitlab urls configuration

#### Ticket Link
[MM-12420](https://mattermost.atlassian.net/browse/MM-12420)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/admin_console/admin_definition.jsx [hunk 1]
```diff
@@ -747,7 +747,7 @@ export default {
                     },
                     onConfigSave: (config) => {
                         const newConfig = {...config};
-                        newConfig.GitLabSettings.UserApiEndpoint = config.GitLabSettings.Url + '/api/v4/user';
+                        newConfig.GitLabSettings.UserApiEndpoint = config.GitLabSettings.Url.replace(/\/$/, '') + '/api/v4/user';
```
Comment (lieut-data) [inline components/admin_console/admin_definition.jsx]: Should we potentially match multiple trailing `/`s here?

Comment (jespino) [reply components/admin_console/admin_definition.jsx]: Good point

Comment (jespino) [reply components/admin_console/admin_definition.jsx]: I'm trying it and there are a lot of weird behaviors on it, for example, I can write `http://localhost:8080/////` and will generate `http://localhost:8080/api/v4/user` but if then I add a "a" at the end, it generates `http://localhost:8080/////a/api/v4/user`. I'm thinking if this is enough to fix the obvious valid problem, and leave the other cases to the user.

**Comment by jespino** [comment]
To break this specific case you have to change how the generated string done. It is done in a very concrete place and specifically for GitLab in this case.

**Comment by jespino** [comment]
cherry-picked to 5.4

**Pull request merged**